src/shared_memory_manager.cc

```diff
@@ -145,6 +146,35 @@ OpenSharedMemoryRegion(const std::string& shm_key, int* shm_fd)
   return nullptr;
 }
 
+TRITONSERVER_Error*
+GetSharedMemoryRegionInfo(
```

Comment:
Design looks solid! My only concern is naming overloading. This file has:
- CUDASharedMemoryInfo (struct)
- SharedMemoryInfo (struct)
- GetMemoryInfo (method)
- GetSharedMemoryRegionInfo (new method)

Would it make sense to name this method something more specific like `GetSharedMemoryRegionSize` to make it more readily distinguishable from the rest? I know a user can read the code to tell the difference and the current name makes it more readily extensible in the future, but just a thought.

Reply:
I didn't choose Size in the name since it may return some other fields like permissions in the near term.

The main distinction of this function is that it's checking info from actual shared memory file object, rather than some internal data structure we setup with user-provided values.

Open to other names though if that helps distinguish the functionality

Reply:
That's what I figured. Carry on then, no issue 🚀

Reply:
Renamed `RegionInfo` to `FileInfo` in hopes for some extra clarity